marlowe-dashboard-client/src/MainFrame/State.purs

```diff
@@ -292,10 +300,36 @@ handleAction (PlayAction (Play.AddNewWallet mTokenName)) = do
   walletNickname <- use (_newWalletDetails <<< _walletNicknameString)
   handleAction AddNewWallet
   for_ mTokenName \tokenName ->
-    Template.handleAction $ Template.SetRoleWallet tokenName walletNickname
+    toTemplate $ Template.handleAction $ Template.SetRoleWallet tokenName walletNickname
 
 -- other play actions
-handleAction (PlayAction playAction) = Play.handleAction playAction
+handleAction (PlayAction playAction) = toPlay $ Play.handleAction playAction
+
+------------------------------------------------------------
+-- Note [dummyState]: In order to map a submodule whose state might not exist, we need
```

Comment:
Also, when I have the time, I want to give it a go at implementing `mapMaybeSubmodule` without requiring this dummy state. It should be possible as the `a` in HalogenM we are using is `Unit`